src/Features/Core/Portable/MakeMethodAsynchronous/AbstractMakeMethodAsynchronousCodeFixProvider.cs

```diff
@@ -93,17 +95,35 @@ private bool IsLikelyEntryPointName(string name, Document document)
             // if it has the 'Async' suffix, and remove that suffix if so.
             var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
             var methodSymbolOpt = semanticModel.GetDeclaredSymbol(node) as IMethodSymbol;
+            var compilation = await document.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+            var knownTypes = new KnownTypes(compilation);
 
             bool isOrdinaryOrLocalFunction = methodSymbolOpt.IsOrdinaryMethodOrLocalFunction();
-            if (isOrdinaryOrLocalFunction && !methodSymbolOpt.Name.EndsWith(AsyncSuffix) && !isEntryPoint)
+            if (isOrdinaryOrLocalFunction
+                && !methodSymbolOpt.Name.EndsWith(AsyncSuffix)
+                && !isEntryPoint
+                && NeedsRename(this, methodSymbolOpt, keepVoid, in knownTypes))
             {
                 return await RenameThenAddAsyncTokenAsync(
-                    keepVoid, document, node, methodSymbolOpt, cancellationToken).ConfigureAwait(false);
+                    keepVoid, document, node, methodSymbolOpt, knownTypes, cancellationToken).ConfigureAwait(false);
             }
             else
             {
                 return await AddAsyncTokenAsync(
-                    keepVoid, document, methodSymbolOpt, node, cancellationToken).ConfigureAwait(false);
+                    keepVoid, document, methodSymbolOpt, knownTypes, node, cancellationToken).ConfigureAwait(false);
+            }
+
+            static bool NeedsRename(AbstractMakeMethodAsynchronousCodeFixProvider @this, IMethodSymbol methodSymbol, bool keepVoid, in KnownTypes knownTypes)
```

Comment:
having a static local function that you pass @this to just seems overwrought.  Can you just make a normal instance method in the type?  Since this is all one feature, it seems totally clear to have a helpe rmethod without needing to make it scoped to this one method.

Reply:
I'm going to keep this _mostly_ so I can see how I feel about it after some time passes.

Reply:
SGTM.